Src/FluentAssertions/DataColumnCollectionAssertionExtensions.cs

```diff
@@ -0,0 +1,233 @@
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public static AndConstraint<GenericCollectionAssertions<DataColumn>> NotHaveSameCount(this GenericCollectionAssertions<DataColumn> assertion, DataColumnCollection otherCollection, string because = "",
+            params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(otherCollection, nameof(otherCollection), "Cannot verify count against a <null> collection.");
+
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .WithExpectation("Expected {context:collection} to not have ")
+                .Given(() => assertion.Subject)
+                .ForCondition(subject => subject is not null)
+                .FailWith("the same count as {0}{reason}, but found <null>.", otherCollection)
+                .Then
+                .Given((subject) => (actual: subject.Count(), expected: otherCollection.Count))
+                .ForCondition(count => count.actual != count.expected)
+                .FailWith("{0} column(s){reason}, but found {1}.", count => count.expected, count => count.actual)
+                .Then
+                .ClearExpectation();
+
+            return new AndConstraint<GenericCollectionAssertions<DataColumn>>(assertion);
+        }
+
+        /// <summary>
+        /// Asserts that the current collection of <see cref="DataColumn"/>s contains a <see cref="DataColumn"/> with the specified <paramref name="expectedColumnName"/> name.
+        /// </summary>
+        /// <param name="expectedColumnName">A name for a <see cref="DataColumn"/> that is expected to be in the <see cref="DataColumnCollection"/>.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public static AndConstraint<GenericCollectionAssertions<DataColumn>> ContainColumnWithName(this GenericCollectionAssertions<DataColumn> assertion, string expectedColumnName, string because = "",
+            params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(expectedColumnName, nameof(expectedColumnName), "Cannot verify that the collection contains a <null> DataColumn.");
+
+            if (assertion.Subject is null)
```

Comment:
Should this also be checking if `assertion.Subject.UnderlyingCollection` is null?

Reply:
In normal usage, `NonGenericCollectionWrapper` only gets created when people write one of these:
```
  dataSet.Tables.Should()...
  dataTable.Columns.Should()...
  dataTable.Rows.Should()...
```
A caller would have to go out of their way to get a `null` reference of type `DataTableCollection`, `DataColumnCollection` or `DataRowCollection`. I've never heard of these collection types being used in a context _other_ than a member of `DataSet` or `DataTable`. Do you think additional code is warranted to guard against this??

Maybe it would make sense to replace the constructor on `NonGenericCollectionWrapper` with a static method, and then if the caller actually supplies `null` in the first place, then it doesn't wrap `null`, it returns an actual `null` reference for the wrapper itself too.

Reply:
History has shown that users of FA do get out of their way 😀

Reply:
The latest commit includes the proposed change, whereby `NonGenericCollectionWrapper` will never wrap `null`, and I also just belatedly added a `Guard.ThrowIfArgumentIsNull` to the actual constructor, so that `NonGenericCollectionWrapper` _can_ never wrap `null`.